### DISCUSSION TITLE
pull_request does not trigger workflow but on_pull_request does

### DISCUSSION THREAD

**Discussion by armando-swarm** [community/community#35897] [Actions] [answered]
I'm trying to get into github actions and am currently trying a workflow where on PR tests will run

For some reason, the workflow will trigger on PR creation/reopening/synch when the event type is `pull_request_target` but if it's `pull_request` no actions are executed. It's as if conditions are not met.

Previously I have defined types, paths, and branches. But in an attempt to troubleshoot the issue I removed everything and just ended up with `pull_request` to see if it really triggers if there are no filters. But the workflow still does not trigger no matter what happens to the PR

```
on: [pull_request]
```

The target branch is in our team's repo and the source branch is my fork of the target branch and there are no merge conflicts

By looking into examples and other questions which involve `pull_request` in some way I don't see anything special configured so I'm not sure why this is not triggering. May I know if there are some configurations that I need to check or if there's anything I need to look at?

EDIT: I can confirm that in my case `pull_request` works only if the branch of the PR is part of the same repo as the target branch. But if the source branch comes from our forks, the workflow is not triggered. I am continuing to investigate/research why

EDIT: checked if the accout where the fork is is under the same organization as the target branch and it is. So i'm not sure if it's an account issue

EDIT: I found that the setting below is disabled. I will talk to the repo owner later to enable it to see if this fixes the issue

```
Run workflows from fork pull requests
This tells Actions to run workflows from pull requests originating from repository forks. Note that doing so will give maintainers of those forks the ability to use tokens with read permissions on the source repository.
```

**Comment by thaisonzdx** [comment]
Did you try?
```
on:
  pull_request:
    branches:
      - main
```

**Reply by armando-swarm** [reply]
yes this did not work

**Comment by airtower-luna** [comment]
Can you link the actual PR? Being able to take a look always make it a lot easier to figure out what's going on.

If not: Is the workflow file on the branch you're trying to merge? For `pull_request` it has to be.

**Reply by armando-swarm** [reply]
Hi! Unfortunately can't because the repos are private. Here's what I did with the specifics to what we're doing removed

```
name: Test
on:
  pull_request:
    types: [opened, edited, synchronize, reopened]
    branches:
      - develop
    paths:
      - 'path/to/file/**'
```

The workflow file exists on the source branch and the target branch of the PR. Source branch always has whatever is on the target branch in our case. 

`If not: Is the workflow file on the branch you're trying to merge? For pull_request it has to be.` <- does this also imply that both source and target branch have to have `pull_request` as the event? I think one of these may have `pull_request_target` and now that you mention it I haven't tried making sure that both source and target have the same event and maybe since its merge commit that is causing the issue

edit: nope that wasn't the issue

**Comment by armando-swarm** [comment, marked as answer]
I learned from  https://securitylab.github.com/research/github-actions-preventing-pwn-requests that it was indeed the setting I mentioned. Allowing workflows from repo forks is the quickest fix since we're a very small group working on a private repo although this may not be a safe solution for larger teams - as the article suggests pull_requests from forks makes the repo vulnerable to hijack
